pkg/theme/text.go

```diff
@@ -0,0 +1,395 @@
+package theme
+
+import (
+	"github.com/gookit/color"
+)
+
+var (
+	// DefaultTextColor is the default text color
+	DefaultTextColor = NewTextStyle(color.FgWhite, 0)
+	// DefaultHiTextColor is the default highlighted text color
+	DefaultHiTextColor = NewTextStyle(color.FgLightWhite, 0)
+
+	// SelectedLineBgColor is the background color for the selected line
+	SelectedLineBgColor = NewTextStyle(0, 0)
+
+	// SelectedRangeBgColor is the background color of the selected range of lines
+	SelectedRangeBgColor = NewTextStyle(0, 0)
+
+	OptionsFgColor = NewTextStyle(0, 0)
+
+	DiffTerminalColor = NewTextStyle(color.FgMagenta, 0)
+
+	FgCyan    = NewTextStyle(color.FgCyan, 0)
+	FgRed     = NewTextStyle(color.FgRed, 0)
+	FgGreen   = NewTextStyle(color.FgGreen, 0)
+	FgBlue    = NewTextStyle(color.FgBlue, 0)
+	FgYellow  = NewTextStyle(color.FgYellow, 0)
+	FgMagenta = NewTextStyle(color.FgMagenta, 0)
+)
+
+type TextColorType uint8
+
+const (
+	NoTextColor = TextColorType(iota)
+	BasicTextColor
+	RGBTextColor
+)
+
+type TextStyle interface {
+	Sprint(a ...interface{}) string
+	Sprintf(format string, a ...interface{}) string
+	SetBold(v bool) TextStyle
+	SetReverse(v bool) TextStyle
+	SetUnderline(v bool) TextStyle
+	SetColor(color color.Color, background bool) TextStyle
+	SetRGBColor(r, g, b uint8, background bool) TextStyle
+	Merge(other TextStyle) TextStyle
+	GetOpts() color.Opts
+}
+
+func NewTextStyle(fg color.Color, bg color.Color, opts ...color.Color) TextStyle {
+	res := BasicTextStyle{}
+
+	if fg != 0 {
+		res.hasFg = true
+		res.style = append(res.style, fg)
+	}
+	if bg != 0 {
+		res.hasBg = true
+		res.style = append(res.style, bg)
+	}
+
+	for _, opt := range opts {
+		switch opt {
+		case color.OpBold, color.OpReverse, color.OpUnderscore:
+			res.style = append(res.style, opt)
+		}
+	}
+
+	return res
+}
+
+type BasicTextStyle struct {
+	hasFg bool
+	hasBg bool
+	style color.Style
+}
+
+func (b BasicTextStyle) Sprint(a ...interface{}) string {
+	return b.style.Sprint(a...)
+}
+
+func (b BasicTextStyle) Sprintf(format string, a ...interface{}) string {
+	return b.style.Sprintf(format, a...)
+}
+
+func (b BasicTextStyle) setOpt(opt color.Color, v bool) BasicTextStyle {
```

Comment:
It would be good to add a test or two for this struct (same with the rgb one) i.e. something like calling setOpt a few times and setColor and then asserting on the output for Sprint

Reply:
Added some tests